Design.md

```diff
@@ -0,0 +1,75 @@
+# Architecture
+![VPP agent 10.000 feet](vpp_agent_10K_feet.png "VPP Agent - 10.000 feet view on the architecture")
+
+Brief description:
+* SFC Controller - renders desired network stitching configuration for multiple agents to the Data Store
+* Control Plane APPs - renders specific VPP configuration for multiple agents to the Data Store
+* Client v1 - Control plane can use the Client v1 (VPP Agent Client v1) for submitting configuration for VPP Agents.
+              The Client v1 is based on generated GO structures from protobuf messages & set of helper methods
+              that generates keys and store the data to key the value Data Store.
+* Data Store (ETCD, Redis, etc.) to:
+  * store the VPP configuration
+  * operational state (network counters & statistics, errors...)
+* VPP vSwitch - privileged container that cross connects multiple VNFs
+* VPP VNF - container that runs VPP that acts as Virtual Network Function 
+* Non VPP VNF - non VPP containers can interact together with VPP containers (see below MEMIFs, VETH)
+* Messaging - AD-HOC events (e.g. link UP/Down)
+
+# Requirements
+VPP Agent was designed with following principal requirements:
+* Modular design with API contract
+* Cloud native
+* Fault tolerant
+* Rapid deployment
+* High performance & minimal footprint
+
+
+## Modular design with API contract
+Code organized as plugins. Each plugin exposes API that. This approach allows 
+to extend the functionality by introducing new plugins and enables integration of plugins by using the API.
+
+## Cloud native
+Assumption: data plane & control plane can be divided to multiple microservices.
+Each microservice is independent therefore normally can occur that the configuration
```

Comment:
Each microservice is independent. Therefore, it may occur that the configuration is incomplete: an object can refer to a non-existing object in configuration.